app/src/test/java/com/moonpig/mvisample/productdetail/ProductDetailViewModelTest.kt

```diff
@@ -15,50 +18,44 @@ class ProductDetailViewModelTest {
     private val productDetailUseCase: ProductDetailUseCase = mock()
     private val productDetailTracker: ProductDetailTracker = mock()
 
+    private val viewModel = givenAProductDetailViewModel()
+    private val viewStateObserver = viewModel.viewState().test()
```

Comment:
I'm not sure about making these class level, I'm surprised all the tests pass, as we're now theoretically binding lots of intents to the `intentSubject` in the `viewModel`. I would have thought this will cause test flakiness as say for whatever reason a previously bound `intent` will emit something, it might cause another test to break as a state was emitted that it didn't expect. So if we do this, we should probably have a setUp and tearDown. 

@tyates what do you think?

Reply:
I have never had any problems with creating the SUT in the properties. 

I think this would cause test coupling in JUNIT3, as far as I know JUNIT4 instantiates a new object for each test. 
Otherwise your mocks would also carry state between tests and they don't.